lib/internal/loader/resolveRequestUrl.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+const { URL } = require('url');
+const internalCJSModule = require('internal/module');
+const internalURLModule = require('internal/url');
+const NativeModule = require('native_module');
+const { extname } = require('path');
+
+const { ModuleWrap } = process.binding('module_wrap');
+
+const search = require('internal/loader/search');
+const asyncReadFile = require('util').promisify(require('fs').readFile);
+const debug = require('util').debuglog('esm');
+
+const ArrayJoin = Function.call.bind(Array.prototype.join);
+const ArrayMap = Function.call.bind(Array.prototype.map);
+
+const { Error } = global;
+
+const createDynamicModule = (exports, url = '', evaluate) => {
+  debug(
+    `creating ESM facade for ${url} with exports: ${ArrayJoin(exports, ', ')}`
+  );
+  const names = ArrayMap(exports, (name) => `${name}`);
+  // sanitized ESM for reflection purposes
+  const src = `export let executor;
```

Comment:
This only works because this is only ever called with `['default']` as its first argument 🤔 

Maybe the export default should be hardcoded for clarity? or does this need to be more generic?

Then again since this is internal only you can document putting `'executor'` in the `exports` array as a "don't do that".